### PR TITLE
Single node clusters and update CNI version in readme

### PR DESCRIPTION
**Pull request by prateekgogia**
Issue #, if available:

Description of changes:
- Update Readme to install latest CNI version
- Add support for running single node guest clusters using KIT

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: operator/pkg/apis/controlplane/v1alpha1/controlplane.go [hunk 1]
```diff
@@ -46,9 +46,10 @@ type ControlPlaneList struct {
 // master and etcd are configured to run. By default, KIT uses all the default
 // values and ControlPlaneSpec can be empty.
 type ControlPlaneSpec struct {
-	KubernetesVersion string     `json:"kubernetesVersion,omitempty"`
-	Master            MasterSpec `json:"master,omitempty"`
-	Etcd              Etcd       `json:"etcd,omitempty"`
+	KubernetesVersion         string     `json:"kubernetesVersion,omitempty"`
+	ColocateAPIServerWithEtcd bool       `json:"colocateAPIServerWithEtcd,omitempty"`
```
Comment (hakuna-matatah) [inline operator/pkg/apis/controlplane/v1alpha1/controlplane.go:50]: what's the use-case for this ?

**Comment by hakuna-matatah** [comment]
Can you update the description and/or title of PR ? I see that you are also adding a single-node functionality as part of this PR outside of updating CNI.

File: operator/pkg/apis/controlplane/v1alpha1/controlplane.go (hunk 1 above)
Comment (prateekgogia) [reply operator/pkg/apis/controlplane/v1alpha1/controlplane.go:50]: There are scenarios where we just want to test the functionality of a component and don't need a 4-5 node cluster.

File: substrate/pkg/utils/discovery/tags.go [hunk 2]
```diff
@@ -27,8 +27,12 @@ const (
 )
 
 func Tags(substrate *v1alpha1.Substrate, name *string) []*ec2.Tag {
+	return OwnerWithNameTag(aws.String(OwnerTagKey), aws.String(substrate.Name), name)
+}
+
+func OwnerWithNameTag(ownerTag, ownerValue, name *string) []*ec2.Tag {
```
Comment (hakuna-matatah) [inline substrate/pkg/utils/discovery/tags.go:33]: why this function ? unlike before ?

File: operator/pkg/controllers/master/kubeapiserver.go [hunk 3]
```diff
@@ -84,7 +84,8 @@ func apiServerPodSpecFor(controlPlane *v1alpha1.ControlPlane) v1.PodSpec {
 		HostNetwork:                   true,
 		DNSPolicy:                     v1.DNSClusterFirstWithHostNet,
 		PriorityClassName:             "system-cluster-critical",
-		NodeSelector:                  nodeSelector(controlPlane.ClusterName()),
+		NodeSelector:                  nodeSelector(controlPlane.ClusterName(), controlPlane.Spec.ColocateAPIServerWithEtcd),
+		Affinity:                      affinity(controlPlane.Spec.ColocateAPIServerWithEtcd),
```
Comment (hakuna-matatah) [inline operator/pkg/controllers/master/kubeapiserver.go:88]: Remind me why do we need Affinity for apiserver component and not for other components like KCM/Scheduler/EncryptionProvider/Authenticator ?

Comment (prateekgogia) [reply operator/pkg/controllers/master/kubeapiserver.go:88]: They are running as daemonsets

File: substrate/pkg/utils/discovery/tags.go (hunk 2 above)
Comment (prateekgogia) [reply substrate/pkg/utils/discovery/tags.go:33]: Trying to remove dependency on *v1alpha1.Substrate in the client library using this function.

**Review by hakuna-matatah — APPROVED** [review]
lgtm

**Pull request merged**